### PR TITLE
HPCC-15572 Allow the parser state table to dynamically expand

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss new PR

File: ecl/hql/hqlgram.y [hunk 1]
```diff
@@ -32,6 +32,9 @@
 //Adding a comment to reference unused parameters also fails to solve it because it ignores references in comments (and a bit ugly)
 //fixing bison to ignore destructor {} for need use is another alternative - but would take a long time to feed into a public build.
 %{
+//The following allows the state table to expand.  It indicates memcpy of the data is valid.
```
Comment (jamienoss) [inline ecl/hql/hqlgram.y]: I think you mean "state stack" rather than "state table".

**Comment by jamienoss** [comment]
@ghalliday are there any bison version dependencies for this flag?

**Comment by ghalliday** [comment]
No idea, but I doubt it.

**Comment by jamienoss** [comment]
@richardkchapman merge at will.

**Pull request merged**